core/src/com/unciv/logic/map/MapUnit.kt

```diff
@@ -657,6 +662,20 @@ class MapUnit {
                 .sumBy { it.params[0].toInt() }
     }
 
+    private fun getTerrainDamage() {
+        // hard coded mountain damage for now
+        if (getTile().baseTerrain == Constants.mountain) {
+            val tileDamage = 50
```

Comment:
takeDamage(50)

Reply:
I did it the same way citadel damage is done from MapUnit.kt:

```
    private fun getCitadelDamage() {
        // Check for Citadel damage
        val applyCitadelDamage = currentTile.neighbors
                .filter { it.getOwner() != null && civInfo.isAtWarWith(it.getOwner()!!) }
                .map { it.getTileImprovement() }
                .filter { it != null && it.hasUnique("Deal 30 damage to adjacent enemy units") }
                .any()

        if (applyCitadelDamage) {
            health -= 30

            if (health <= 0) {
                civInfo.addNotification("An enemy [Citadel] has destroyed our [$name]", currentTile.position, Color.RED)
                // todo - add notification for attacking civ
                destroy()
            } else civInfo.addNotification("An enemy [Citadel] has attacked our [$name]", currentTile.position, Color.RED)
        }
    } 
```

As far as I'm aware, takeDamage isn't available to the MapUnit class. Should terrain damage be put somewhere else?

Reply:
There isn't?!
Huh
Maybe it's in the combattant class
K then